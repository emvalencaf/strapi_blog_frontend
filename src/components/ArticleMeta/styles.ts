import styled, { css } from "styled-components";

export const Wrapper = styled.footer`
	${({ theme }) => css`
		font-size: ${theme.font.sizes.small};
		color: ${theme.colors.darkerGray};

		.categories span::after {
			content: ", ";
		}

		.categories span:last-child::after {
			content: "";
		}

		a {
			color: ${theme.colors.secondary};
			text-decoration: none;
			transition: : all 350ms ease-in-out;

			&:hover {
				filter: brightness(50%);
			}
		}


	`}
`;
