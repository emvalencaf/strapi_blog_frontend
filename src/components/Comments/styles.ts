import styled, { css } from "styled-components";

export const Wrapper = styled.div`
	${({ theme }) => css`
		max-width: ${theme.sizes.content};
		width: 100%;
		margin: ${theme.spacings.large} 0;
		padding: 0 ${theme.spacings.large};
	`}
`;
