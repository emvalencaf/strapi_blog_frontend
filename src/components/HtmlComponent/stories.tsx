import { Meta, Story } from "@storybook/react/types-6-0";

// components and it's type props
import HtmlComponent, { HtmlComponentProps } from ".";

export default {
  title: "TextComponent",
  component: HtmlComponent,
  args: {
    html: `
      Lorem ipsum, dolor sit amet consectetur adipisicing
      elit. Nihil, iure quaerat! Voluptates ducimus exercitationem,
      reprehenderit laborum, libero cum porro necessitatibus sint incidunt,
      praesentium ipsa quaerat debitis delectus nemo aliquid! Facere.
    `,
  },
  argTypes: {
    children: { type: "string" },
  },
} as Meta;

export const Template: Story<HtmlComponentProps> = (args) => (
  <div>
    <HtmlComponent {...args} />
  </div>
);
