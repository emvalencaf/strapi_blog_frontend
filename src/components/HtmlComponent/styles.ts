import styled, { css } from "styled-components";

export const Container = styled.div`
	${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        max-width: 100%;
        float: none;
      }
    }

    font-size: ${theme.font.sizes.medium};


    p {
      margin: ${theme.spacings.xlarge} 0;
    }

    a, a:visited, a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 350ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    pre {
      font-size: ${theme.font.sizes.small};
      background-color: ${theme.colors.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white}
      margin: ${theme.spacings.xlarge};
      width: 100%;
      overflow-x: auto;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.xlarge} 0;
    }

    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }


    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacings.medium};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid $${theme.colors.mediumGray};
    }

    ul, ol {
      margin: ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;

    }

    table {
      width: 100%;
      border-collapse: collapse;
    }

    .table td, .table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }
  `}
`;
