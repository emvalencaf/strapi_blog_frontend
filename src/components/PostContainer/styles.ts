import styled, { css, DefaultTheme } from "styled-components";

// types
type PostContainerStyledProps = {
	size: "max" | "content";
};

const postContainerStyles = {
	max: (theme: DefaultTheme) => css`
		max-width: ${theme.sizes.max};
	`,
	content: (theme: DefaultTheme) => css`
		max-width: ${theme.sizes.content};
	`,
};

export const PostContainerStyled = styled.div<PostContainerStyledProps>`
	${({ theme, size }) => css`
		width: 100%;
		margin: 0 auto;
		padding: 0 ${theme.spacings.large};
		${postContainerStyles[size](theme)}
	`}
`;
