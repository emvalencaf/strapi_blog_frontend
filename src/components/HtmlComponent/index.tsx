import * as Styled from "./styles";

export type HtmlComponentProps = {
	html: string;
};

const HtmlComponent = ({ html }: HtmlComponentProps) => {
	return <Styled.Container dangerouslySetInnerHTML={{ __html: html }} />;
};

export default HtmlComponent;
