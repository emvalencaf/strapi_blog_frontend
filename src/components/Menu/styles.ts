import styled, { css, DefaultTheme } from "styled-components";

// imported styled
import { Title as HeadingStyles } from "../Heading/styles";

// types
type MenuBehaviorParams = {
	menuVisible: boolean;
	theme: DefaultTheme;
};

const wrapperChanger = (menuVisible: boolean, theme: DefaultTheme) => css`
	@media ${theme.media.lteMedium} {
		left: ${menuVisible ? "0" : "-32rem"};
	}

	left: ${menuVisible ? "0" : "-30rem"};
	overflow-y: ${menuVisible ? "auto" : "hidden"};
`;

const buttonChanger = (menuVisible: boolean, theme: DefaultTheme) => css`
	@media ${theme.media.lteMedium} {
		left: ${menuVisible ? "26rem" : "1rem"};
	}

	left: ${menuVisible ? "26rem" : "-1rem"};
	color: ${menuVisible ? theme.colors.secondary : theme.colors.white};
`;

export const Wrapper = styled.div<MenuBehaviorParams>`
	${({ theme, menuVisible }) => css`
		background: ${theme.colors.primary};
		padding: ${theme.spacings.large};
		display: flex;
		position: fixed;
		z-index: 1;
		width: 100%;
		max-width: 32rem;
		height: 100vh;
		top: 0;
		left: 0;
		transition: all 350ms ease-in-out;
		${wrapperChanger(menuVisible, theme)}
	`}
`;

export const Nav = styled.nav`
	${() => css`
		margin: auto;
		width: 100%;
	`}
`;

export const Logo = styled.div`
	${({ theme }) => css`
		${HeadingStyles} {
			display: flex;
			justify-content: center;
			margin: 0;
			margin-bottom: ${theme.spacings.xlarge};

			img {
				border: 0.5rem solid ${theme.colors.secondary};
			}
		}
	`}
`;

export const OpenClose = styled.a<MenuBehaviorParams>`
	${({ theme, menuVisible }) => css`
		position: fixed;
		top: ${theme.spacings.medium};
		color: ${theme.colors.white};
		background: ${theme.colors.primary};
		z-index: 2;
		width: 3rem;
		height: 3rem;
		left: 26rem;
		transition: all 350ms ease-in-out;
		${buttonChanger(menuVisible, theme)}
	`}
`;
