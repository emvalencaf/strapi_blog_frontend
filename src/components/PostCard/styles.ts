import styled, { css } from "styled-components";

import { Title as HeadingStyles } from "../Heading/styles";

export const Wrapper = styled.div`
	${({ theme }) => css`
		${HeadingStyles} {
			margin-left: 0;
			margin-right: 0;
			margin-top: calc(${theme.spacings.small} - 1.4rem);
			margin-bottom: calc(${theme.spacings.small} - 1rem);
		}

		a {
			text-decoration: none;
			color: inherit;
			transition: all 350ms ease-in-out;
		}

		&:hover a {
			color: ${theme.colors.secondary};
		}

		&:hover img {
			opacity: 0.8;
		}
	`}
`;

export const Cover = styled.img`
	${() => css`
		max-width: 100%;
		transition: opacity 350ms ease-in-out;
	`}
`;

export const Excerpt = styled.p``;
