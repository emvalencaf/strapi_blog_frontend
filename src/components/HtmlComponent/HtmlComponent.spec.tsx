import { renderTheme } from "../../styles/render-theme";
import HtmlComponent from ".";
import { screen } from "@testing-library/react";

describe("<HtmlComponent />", () => {
  it("should render a html", () => {
    renderTheme(<HtmlComponent html={'<b>Children</b>'} />);
    const element = screen.getByText('Children');
    expect(element).toBeInTheDocument();
  });

  it("should match a snapshot", () => {
    const { container } = renderTheme(<HtmlComponent html={'<b>Children</b>'} />);
    expect(container).toMatchSnapshot();
  });
});
