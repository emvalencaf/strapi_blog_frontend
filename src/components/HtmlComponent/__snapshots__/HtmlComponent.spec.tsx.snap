// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlComponent /> should match a snapshot 1`] = `
.c0 {
  font-size: 2.4rem;
}

<div>
  <div
    class="c0"
  >
    <b>
      Children
    </b>
  </div>
</div>
`;
